### PR TITLE
Add migration to apply update to es user mapping

### PR DESCRIPTION
**Pull request by Charl1996**
## Product Description
No visible changes

## Technical Summary
This PR is related to [this PR](https://github.com/dimagi/commcare-hq/pull/31667) which was supposed to update the ES user index by adding a new field, however, it was found that it did not actually update the index. After consulting with Amit, he pointed me to the new approach of updating index mappings through Django migrations (using the `make_elastic_migration` command), hence this PR.

After this PR is done I will proceed to populate the newly added field(s) with the appropriate data by means of [this PR](https://github.com/dimagi/commcare-hq/pull/32864).

### Safety story
Tested locally and on staging.

### Migrations
- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions
- [ ] This PR can be reverted after deploy with no further considerations
I don't think this can be reverted.

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AmitPhulera — APPROVED** [review]

File: corehq/apps/es/migrations/0003_add_assigned_location_ids.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('es', '0002_add_tombstones'),
+    ]
+
+    operations = [
+        corehq.apps.es.migration_operations.UpdateIndexMapping(
```
Comment (millerdev) [inline corehq/apps/es/migrations/0003_add_assigned_location_ids.py:14]: @AmitPhulera is `ptop_preindex` required after this is applied? If yes and if it is not run automatically, we should have the ES migration operations request a reindex after migrations are applied, similar to how [`RequestReindex`](https://github.com/dimagi/commcare-hq/blob/8e7b380c908368492cdfb7742627ad9062052d20/corehq/preindex/django_migrations.py#L78) does.

Comment (AmitPhulera) [reply corehq/apps/es/migrations/0003_add_assigned_location_ids.py:14]: No. `ptop_preindex` is not required after this is applied. The relevant docs would be indexed again by the migration added by Charl in one of the previous commits.

Comment (millerdev) [reply corehq/apps/es/migrations/0003_add_assigned_location_ids.py:14]: @AmitPhulera I should have asked one more question: what do you mean by "previous commits"? Was that on a different PR? I only see one commit on this PR.

Comment (Charl1996) [reply corehq/apps/es/migrations/0003_add_assigned_location_ids.py:14]: @millerdev I think @AmitPhulera might be referring to [this PR](https://github.com/dimagi/commcare-hq/pull/32864).

**Pull request merged**